# TEAM GMM PROJECT
*Lukas Elenbergas, **1712238***

*Team Name: **Solo Mission***

*Task: **Music Generation***
https://www.kaggle.com/datasets/googleai/musiccaps

# MusicCaps - Dataset

MusicCaps dataset implementation based on: https://github.com/nateraw/download-musiccaps-dataset

In [3]:
# ---------------------------------------------------
# MUSICCAPS DATASET INIT
# ---------------------------------------------------
from Helpers import download_musiccaps_modified

music_caps_dataset = download_musiccaps_modified.main(
    'C:\\Projects\\GMM\\MusicData',        
    num_proc=16,
)
# ---------------------------------------------------

In [2]:
print(music_caps_dataset[0])

{'ytid': '-0Gj8-vB1q4', 'start_s': 30, 'end_s': 40, 'audioset_positive_labels': '/m/0140xf,/m/02cjck,/m/04rlf', 'aspect_list': "['low quality', 'sustained strings melody', 'soft female vocal', 'mellow piano melody', 'sad', 'soulful', 'ballad']", 'caption': 'The low quality recording features a ballad song that contains sustained strings, mellow piano melody and soft female vocal singing over it. It sounds sad and soulful, like something you would hear at Sunday services.', 'author_id': 4, 'is_balanced_subset': False, 'is_audioset_eval': True, 'audio': {'path': 'C:\\Projects\\GMM\\MusicData\\-0Gj8-vB1q4.wav', 'array': array([-0.00193254,  0.00109734,  0.00309335, ..., -0.01940443,
       -0.02377406,  0.        ]), 'sampling_rate': 44100}, 'download_status': True}


# How a dataset sample looks:

{

        'ytid': '-0Gj8-vB1q4'
        'start_s': 30, 
        'end_s': 40, 
        'audioset_positive_labels': '/m/0140xf,/m/02cjck,/m/04rlf', 
        'aspect_list': "[
            'low quality', 
            'sustained strings melody', 
            'soft female vocal', 
            'mellow piano melody', 
            'sad', 
            'soulful', 
            'ballad'
            ]", 
        'caption': 'The low quality recording features a ballad song that contains sustained strings, mellow piano melody and soft female vocal singing over it. It sounds sad and soulful, like something you would hear at Sunday services.', 
        'author_id': 4, 
        'is_balanced_subset': False, 
        'is_audioset_eval': True, 
        'audio': 
        {
            'path': 'C:\\Projects\\GMM\\MusicData\\-0Gj8-vB1q4.wav', 
            'array': array([-0.00193254,  0.00109734,  0.00309335, ..., -0.01940443,-0.02377406,  0.]),
            'sampling_rate': 44100
        }, 
        'download_status': True
        
}

In [ ]:
# ---------------------------------------------------
# SAMPLE EXPLORATION VIA GRADIO
# ---------------------------------------------------
import gradio as gr

def get_example(idx):
    ex = music_caps_dataset[idx]
    return ex['audio']['path'], ex['caption']

gr.Interface(
    get_example,
    inputs=gr.Slider(0, len(ds) - 1, value=0, step=1),
    outputs=['audio', 'textarea'],
    live=True
).launch()
# ---------------------------------------------------

# MusicGen - Sample Generation

Using the AudioCraft library for the MusicGen model: https://github.com/facebookresearch/audiocraft


In [7]:
# ---------------------------------------------------
#
# ---------------------------------------------------
from audiocraft.models import MusicGen

# ---------------------------------------------------

ImportError: cannot import name 'ModelMetaclass' from 'pydantic.main' (c:\ProgramData\anaconda3\envs\GMM-TEAM\lib\site-packages\pydantic\main.py)

# OpenAI GPT-3.5 - Band Description Generation

In [ ]:
# ---------------------------------------------------
#
# ---------------------------------------------------

# ---------------------------------------------------